vividus-plugin-web-app/src/main/java/org/vividus/selenium/WebDriverFactory.java

```diff
@@ -71,35 +79,57 @@
     private IJsonUtils jsonUtils;
 
     private final Supplier<DesiredCapabilities> seleniumGridDesiredCapabilities = Suppliers.memoize(
-        () -> new DesiredCapabilities(propertyParser.getPropertyValuesByPrefix(SELENIUM_GRID_PROPERTY_PREFIX)));
+        () -> getCapabilitiesByFamily(SELENIUM_GRID_PROPERTY_PREFIX));
 
-    private final ConcurrentHashMap<WebDriverType, WebDriverConfiguration> configurations = new ConcurrentHashMap<>();
+    private final Supplier<DesiredCapabilities> localDriverDesiredCapabilities = Suppliers.memoize(
+        () -> getCapabilitiesByFamily(LOCAL_DRIVER_PROPERTY_PREFIX));
 
-    @Override
-    public WebDriver getWebDriver(DesiredCapabilities desiredCapabilities)
+    private final LoadingCache<Boolean, DesiredCapabilities> webDriverCapabilities = CacheBuilder.newBuilder()
+            .build(new CacheLoader<Boolean, DesiredCapabilities>()
+            {
+                public DesiredCapabilities load(Boolean local)
+                {
+                    DesiredCapabilities localCapabilities = localDriverDesiredCapabilities.get();
+                    if (local == true)
+                    {
+                        return localCapabilities;
+                    }
+                    return localCapabilities.merge(seleniumGridDesiredCapabilities.get());
+                }
+            });
+
+    private final Map<WebDriverType, WebDriverConfiguration> configurations = new ConcurrentHashMap<>();
+
+    private DesiredCapabilities getCapabilitiesByFamily(String prefix)
     {
-        return getWebDriver(webDriverType, desiredCapabilities);
+        return propertyParser.getPropertyValuesByPrefix(prefix)
+            .entrySet()
+            .stream()
+            .map(e -> equalsAnyIgnoreCase(e.getValue(), "true", "false")
```

Comment:
can we reuse https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/BooleanUtils.html#toBooleanObject-java.lang.String- here?

Reply:
It will cause unpredictable side effects
![image](https://user-images.githubusercontent.com/17816729/76729359-08335e80-676a-11ea-80fb-60ac9052b1ec.png)